pkg/semtech-loramac/contrib/semtech_loramac.c

```diff
@@ -609,7 +609,7 @@ void *_semtech_loramac_event_loop(void *arg)
                 case MSG_TYPE_MAC_TIMEOUT:
                 {
                     DEBUG("[semtech-loramac] MAC timer timeout\n");
-                    void (*callback)(void) = msg.content.ptr;
+                    void (*callback)(void) = (void (*)(void))(uintptr_t)msg.content.ptr;
```

Comment:
This is actually problematic. On the AVR the program counter is up to 22 bit wide, while addresses in RAM are always 16 bit wide.

(In the DEBUG print in the WIFI driver above it would also be problematic, but a broken debug message is nothing I personally care much about. Here, it would result in a broken application.)

But maybe this is something better suited for a follow up PR, as this is not really related with the issue this PR wants to address and the issue is not introduced by this PR.